### PR TITLE
CORE-14632: Add flow driver to template

### PR DESCRIPTION
**Pull request by tlawson3**
Adding the new flow driver to the template project, giving us the ability to write automated workflow tests
The gradle config is using the minimalist approach, I plan to evaluate the benefits of using a convention plugin under ticket CORE-16396
Updated Corda artifacts to 5.1 versions

### CODE REVIEW

**Review by christian-napoles-r3 — APPROVED** [review]

File: workflows/src/test/java/com/r3/developers/serializers/MemberX500NameSerializer.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import net.corda.v5.base.types.MemberX500Name;
+
+import java.io.IOException;
+
+public class MemberX500NameSerializer extends JsonSerializer<MemberX500Name> {
+    @Override
+    public void serialize(MemberX500Name value, JsonGenerator generator, SerializerProvider serializers) throws IOException {
+        generator.writeString(value.toString());
```
Comment (yarivus) [inline workflows/src/test/java/com/r3/developers/serializers/MemberX500NameSerializer.java:13]: where is the serialized output stored ?

**Review by yarivus — COMMENTED** [review]
see question

File: workflows/src/test/java/com/r3/developers/serializers/MemberX500NameSerializer.java (hunk 1 above)
Comment (tlawson3) [reply workflows/src/test/java/com/r3/developers/serializers/MemberX500NameSerializer.java:13]: We're overriding a function from Jackson > JsonSerializer which handles the generator and the value it produces internally

**Review by yarivus — APPROVED** [review]
LGTM

**Comment by tlawson3** [comment]
Regpack tests updated and passed: 
https://ci02.dev.r3.com/job/QA/job/Craft_Tests/job/qa-regpack-csde/job/tlawson%252FCORE-14632%252FupgradeTo5-1/3/

**Pull request merged**